src/clj/puppetlabs/services/jruby/jruby_core.clj

```diff
@@ -282,3 +289,50 @@
       (cli-ruby! config
         (concat ["-e" (format "load '%s'" url) "--"] args))
       (log/errorf "command %s could not be found in %s" command bin-dir))))
+
+(defmacro with-jruby-instance
+  "Encapsulates the behavior of borrowing and returning a JRubyInstance.
+  Example usage:
+
+  (let [pool-manager-service (tk-app/get-service app :PoolManagerService)
+        pool-context (pool-manager/create-pool pool-manager-service config)]
+    (with-jruby-instance
+      jruby-instance
+      pool-context
+      event-callbacks
+      reason
+
+      (do-something-with-a-jruby-instance jruby-instance)))
+
+  Will throw an IllegalStateException if borrowing a JRubyInstance times out."
+  [jruby-instance pool-context reason event-callbacks & body]
```

Comment:
I guess I was thinking that we'd still have a `register-event-callback` fn in `core`, and people would call that, passing in the pool and the callback.  So, not on the config.  Would that work?